org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServiceAccessor.java

```diff
@@ -465,6 +465,20 @@ private static LanguageServerWrapper getLSWrapper(@Nullable IProject project,
 		}
 	}
 
+	public static LanguageServerWrapper startLanguageServer(@NonNull LanguageServerDefinition serverDefinition) throws IOException {
+		synchronized (startedServers) {
+			LanguageServerWrapper wrapper = startedServers.stream().filter(w -> w.serverDefinition == serverDefinition).findFirst().orElse(null);
+			if (wrapper == null) {
+				wrapper = new LanguageServerWrapper(serverDefinition, null);
```

Comment:
This looks good to me and I would be happy to merge, but I was wondering if we could do better than `orElse(null)`. Could we not just use `new LanguageServerWrapper(serverDefinition, null)` in the `orElse()` above and then just call `startedServers.add(wrapper)`, the startedServers is a set so even if the wrapper is already in it, nothing bad should happen. If we want to avoid the add when the value was already present, we could use `orElseGet()`.

What do you think about that?